models/port.go

```diff
@@ -20,6 +21,9 @@ func (ports *Ports) Parse(ps []core_v1.ServicePort) {
 func (port *Port) Parse(p core_v1.ServicePort) {
 	port.Name = p.Name
 	port.Protocol = string(p.Protocol)
+	if p.AppProtocol != nil {
```

Comment:
I am having a doubt in here about what's better:
1) Convert the *string to string in here and do the if statement to check the if (as you have now), or
2) Move the Port.AppProtocol field to *string and let the `json.Marshal` ([here](https://github.com/kiali/kiali/blob/87ec70e9309b04b4439b95882b516b1c0ee6ba39/handlers/base.go#L13-L23)) convert the nil to string (which is nil).

Something like this:
```golang
type Port struct {
    AppProtocol *string `json:"appProtocol,omitempty"`
}

func (port *Port) Parse(p core_v1.ServicePort) {
	port.Name = p.Name
	port.Protocol = string(p.Protocol)
        port.Port = p.Port
        port.AppProtocol = p.AppProtocol
}
```

I am happy with both ways. What do you think?

Reply:
+1 for the second solution